### ISSUE TITLE
vim mode Esc gets captured by tooltip

### ISSUE THREAD

**Issue by stefanoborini** [rstudio/rstudio#5250]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1335
    OS Version      : macOS 10.14.6
    R Version       : not relevant

### Steps to reproduce the problem

Annoyance with the vim mode enabled. When in edit mode, if a tooltip about e.g. function signature is shown, the subsequent Esc does not leave edit mode for command mode. Instead, it is captured by the tooltip as a closeEvent, leaving the editor in insert mode.

I've written :wq pretty much all over the code.

As a workaround I disabled the tooltips.

**Comment by ronblum** [comment]
@stefanoborini I can understand the annoyance of having to keep switching between having to hit <kbd>Esc</kbd> twice vs. once.  I'll mark this is as an enhancement to review as we continue development of RStudio. That said, do you have any suggestions for alternatives? Thanks!

**Comment by stefanoborini** [comment]
Not really. Honestly I am a bit surprised it happens because if I understand correctly, RStudio uses Qt, and Qt tooltips are not supposed to receive keyboard events, but I might be wrong.

**Comment by jmcphers** [comment]
>  RStudio uses Qt, and Qt tooltips are not supposed to receive keyboard events, but I might be wrong.

You're probably right, but these are HTML tooltips!

**Comment by mskar** [comment]
I think the default behavior should be to always press escape only once to exit insert mode.
That is the default behavior in PyCharm and VS code. 

I will never ever simply want to hide the tooltip.

I imagine that some people want to hide the tooltip so that they start a newline by pressing Enter.
If I want to start a new line with the tooltip open, I can press Shift Enter (this already works in PyCharm and RStudio) or Escape o.

**Comment by ChrisKeefe** [comment]
Another user of RStudio's vim keybinds here. I do most of my development with the Vim extension for VSCode, and personally prefer the RStudio behavior to the suggested behavior. 
> I will never ever simply want to hide the tooltip.

Tooltips are useful, but I close them frequently while actively editing, as they often obstruct code that provides useful context. I get enough value from them that I prefer to keep them enabled, but `<Esc>` makes it easy to clear them and carry on. 

> I think the default behavior should be to always press escape only once to exit insert mode.

Despite my personal preferences, I am in favor of this recommendation, because alignment with other popular editors reduces friction for people moving to RStudio.

Two keybindings might be useful here, with `<Esc>` closing the tooltip and returning to normal mode, and `<Shift><Esc>` closing the tooltip and remaining in insert mode.

**Comment by mskar** [comment]
Your two shortcut suggestion is great, @ChrisKeefe.
The 2 bindings you described work by default in VS code.
I agree that 
> alignment with other popular editors reduces friction for people moving to RStudio.
but it also makes it easier to switch back and forth between editors, as I often do.

I summed up the suggestions so far in a table:
```
| Keybinding   | Action                                      | IDE                      |
|--------------|---------------------------------------------|--------------------------|
| Enter        | Select tooltip suggestion                   | RStudio, VScode, PyCharm |
| Shift Enter  | Close the tooltip and start a new line      | RStudio, PyCharm         |
| Escape       | Close the tooltip and return to normal mode | VScode, PyCharm          |
| Shift Escape | Close the tooltip and stay in insert mode   | VScode                   |
```
The IDE columns lists IDE that have the desired binding/behavior by default.

**Comment by ChrisKeefe** [comment]
> The 2 bindings you described work by default in VS code.

You learn something new every day. Thanks @mskar!

**Comment by stefanoborini** [comment]
@ChrisKeefe tooltips should never be closed manually and should never intercept and consume events. It's a bug that a tooltip receives events. What you propose is a violation of a well established convention in UI design.

**Comment by stefanoborini** [comment]
A bug should not be closed if it exists and it's still present.

**Comment by mskar** [comment]
This issue is still relevant in RStudio Version 1.4.1103.

**Comment by muhark** [comment]
Sorry to necrobump--has a fix been found for this?
